diesel/src/pg/types/primitives.rs

```
@@ -21,6 +21,22 @@ impl ToSql<sql_types::Bool, Pg> for bool {
     }
 }
 
+#[cfg(feature = "postgres_backend")]
+impl FromSql<sql_types::CChar, Pg> for char {
+    fn from_sql(bytes: PgValue<'_>) -> deserialize::Result<Self> {
+        Ok(bytes.as_bytes()[0] as char)
+    }
+}
+
+#[cfg(feature = "postgres_backend")]
+impl ToSql<sql_types::CChar, Pg> for char {
+    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Pg>) -> serialize::Result {
+        out.write_all(&[*self as u8])
```

This should return an error if the `char` is not representable by a `u8`